xfn-gotemplate/Dockerfile

```diff
@@ -0,0 +1,15 @@
+# Compile stage
+FROM golang:1.19 AS build-env
+
+WORKDIR /build
+COPY . /build
+
+RUN go build -o bin/fn main.go
+
+FROM ubuntu:22.04
```

Comment:
One of the best practices we should probably document is to use something like https://github.com/GoogleContainerTools/distroless or Alpine where possible. In fact if we could have our own build tooling use a fairly small set of base images it would keep pulls fast and the cache small.

That said, I'm pleased that this indicates we're able to pull and unpack slightly larger and presumably more complicated OCI image.

Reply:
Updated to use same base image/version as core XP which is `gcr.io/distroless/static@sha256:11364b4198394878b7689ad61c5ea2aae2cd2ed127c09fc7b68ca8ed63219030 `